includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -118,7 +118,7 @@ public function is_no_such_source_error( $error ) {
 		return (
 			$error &&
 			'invalid_request_error' === $error->type &&
-			preg_match( '/(No such source|No such payment)/i', $error->message )
+			preg_match( '/No such (source|PaymentMethod)/i', $error->message )
```

Comment:
This is a commit of my suggestion in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1475#discussion_r626136251